packages/nextjs/hooks/scaffold-eth/useScaffoldEventRead.ts

```diff
@@ -0,0 +1,124 @@
+import { ContractAbi, ContractName } from "~~/utils/scaffold-eth/contract";
+import { ExtractAbiEventNames } from "abitype";
+import { useProvider, useContract } from "wagmi";
+import { useDeployedContractInfo } from "~~/hooks/scaffold-eth";
+import { ethers } from "ethers";
+import { useEffect, useState } from 'react';
+
+/**
+ * @dev reads events from a deployed contract
+ * @param config - The config settings
+ * @param config.contractName - deployed contract name
+ * @param config.eventName - name of the event to listen for
+ * @param config.fromBlock - the block number to start reading events from (default: 1000 blocks before current block)
+ * @param config.filters - filters to be applied to the event (parameterName: value)
+ * @param config.blockData - if set to true it will return the block data for each event (default: false)
+ * @param config.transactionData - if set to true it will return the transaction data for each event (default: false)
+ * @param config.receiptData - if set to true it will return the receipt data for each event (default: false)
+ */
+export const useScaffoldEventRead = <
```

Comment:
Would `useScaffoldEventHistory` make more sense here? Not entirely sure, but that might make the distinction between this hook and the `useScaffoldEventSubscriber` clearer I think.

Reply:
> Would useScaffoldEventHistory make more sense here? Not entirely sure, but that might make the distinction between this hook and the useScaffoldEventSubscriber clearer I think.

++, `useScaffoldEventHistory` love this name !!

Reply:
Changed!